minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/service/MiNiFiCommandSender.java

```diff
@@ -0,0 +1,133 @@
+        this.miNiFiParameters = miNiFiParameters;
+    }
+
+    public Optional<String> sendCommand(String cmd, Integer port) throws IOException {
+        Optional<String> response = Optional.empty();
+
+        if (port == null) {
+            LOGGER.info("Apache MiNiFi is not currently running");
+            return response;
+        }
+
+        try (Socket socket = new Socket()) {
+            LOGGER.debug("Connecting to MiNiFi instance");
+            socket.setSoTimeout(10000);
+            socket.connect(new InetSocketAddress("localhost", port));
+            LOGGER.debug("Established connection to MiNiFi instance.");
+
+            LOGGER.debug("Sending {} Command to port {}", cmd, port);
+            OutputStream out = socket.getOutputStream();
+            out.write((cmd + " " + miNiFiParameters.getSecretKey() + "\n").getBytes(StandardCharsets.UTF_8));
+            out.flush();
+            socket.shutdownOutput();
+
+            InputStream in = socket.getInputStream();
+            StringBuilder sb = new StringBuilder();
+            int numLines = 0;
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(in))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    if (numLines++ > 0) {
+                        sb.append("\n");
+                    }
+                    sb.append(line);
+                }
+            }
+
+            String responseString = sb.toString().trim();
+
+            LOGGER.debug("Received response to {} command: {}", cmd, responseString);
+            response = Optional.of(responseString);
+        }
+        return response;
+    }
+
+    <T> T sendCommandForObject(String cmd, Integer port, String... extraParams) throws IOException {
+        T response;
+        try (Socket socket = new Socket("localhost", port)) {
+            OutputStream out = socket.getOutputStream();
+            String argsString = Arrays.stream(extraParams).collect(Collectors.joining(" ", " ", ""));
+            String commandWithArgs = cmd + " " + miNiFiParameters.getSecretKey() + argsString + "\n";
+            out.write((commandWithArgs).getBytes(StandardCharsets.UTF_8));
+            LOGGER.debug("Sending {} command to MiNiFi with the following args: [{}]", cmd, argsString);
+            out.flush();
+
+            socket.setSoTimeout(5000);
+            InputStream in = socket.getInputStream();
+
+            ObjectInputStream ois = new ObjectInputStream(in);
+            Object o = ois.readObject();
+            ois.close();
+            out.close();
+            response = castResponse(cmd, o);
+        } catch (EOFException | ClassNotFoundException | SocketTimeoutException e) {
+            String message = "Failed to get response for " + cmd + " Potentially due to the process currently being down (restarting or otherwise). Exception:" + e;
+            LOGGER.error(message);
```

Comment:
The exception should be passed as a separate argument instead of being concatenated to the message string:
```suggestion
            String message = "Failed to get response for " + cmd + " Potentially due to the process currently being down (restarting or otherwise)";
            LOGGER.error(message, e);
```